cmd/castRunner.go

```diff
@@ -0,0 +1,356 @@
+	cmdContext := fmt.Sprintf("kube-context=%s", kubeContext)
+	ksCmd := exec.CommandContext(ctx, ksPath, "tap", "-n", namespace, "--set", "headless=true", "--set", cmdConfig, "--set", cmdContext)
+
+	// k communicates kubeshark tap errors
+	k := make(chan error)
+	go func() {
+		err := ksCmd.Run()
+		k <- err
+	}()
+
+	go func() {
+		err := <-k
+		if err != nil && err.Error() != "signal: killed" {
+			log.WithError(err).Info("Error running kubeshark tap.")
+			cancel()
+		}
+	}()
+
+	log.Info("Started kubeshark tap. ")
+
+	// Helm deploy cast
+
+	// Define a public chart repository.
+	chartRepo := repo.Entry{
+		Name: "corshatech",
+		URL:  "https://corshatech.github.io/cast/",
+	}
+
+	// Add a chart-repository to the client.
+	if err := helmClient.AddOrUpdateChartRepo(chartRepo); err != nil {
+		log.WithError(err).Info("Error adding CAST helm repo.")
+		cancel()
+	}
+
+	log.Info("Added CAST helm chart repository.")
+
+	// Install CAST chart release.
+
+	defer castCleanup(helmClient, clientset, ksPath)
+
+	go func(ctx context.Context) {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			deployCast(ctx, cancel, helmClient, clientset, config)
+		}
+	}(ctx)
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+
+	// wait for signal interrupt
+	select {
+	case <-ctx.Done():
+	case <-c:
+		cancel()
+	}
+
+}
+
+// deployCast is a helper function that deploys the CAST helm chart.
+func deployCast(ctx context.Context, cancel context.CancelFunc, helmClient helm.Client, clientset *kubernetes.Clientset, config *rest.Config) {
+
+	chartSpec := helm.ChartSpec{
+		ReleaseName:     "cast",
+		CreateNamespace: true,
+		ChartName:       "corshatech/cast",
+		Namespace:       "cast",
+		UpgradeCRDs:     true,
+		Wait:            true,
+		Timeout:         2 * time.Minute, // postgres pod can take >30s to be ready.
+	}
+	if _, err := helmClient.InstallOrUpgradeChart(context.Background(), &chartSpec, nil); err != nil {
+		handleError("Error installing CAST helm chart.", err, ctx)
+		cancel()
+		return
+	}
+
+	log.Info("CAST release is ready.")
+
+	// Portforward UI pod to localhost:3000
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=cast-ui",
+	}
+	pods, err := clientset.CoreV1().Pods("cast").List(context.Background(), listOptions)
+	if err != nil {
+		handleError("Error listing pods in cast namespace.", err, ctx)
+		cancel()
+	}
+
+	if len(pods.Items) != 1 && ctx.Err() == nil {
+		log.Infof("More than 1 pod found under 'cast-ui' label. Pods: %v", pods.Items)
+	}
+
+	uiPod := pods.Items[0]
+
+	// stopCh controls the port forwarding lifecycle. When it gets closed the
+	// port forward will terminate
+	stopCh := make(chan struct{}, 1)
+	// readyCh communicates when the port forward is ready to get traffic
+	readyCh := make(chan struct{})
+
+	go portForward(uiPod, config, stopCh, readyCh) //nolint:errcheck
+	select {
+	case <-readyCh:
+		println("Port forwarding is ready. CAST is available at http://localhost:3000/.")
```

Comment:
Might be easier for terminals to hyperlink this if we remove the trailing dot
```suggestion
		println("Port forwarding is ready. CAST is available at http://localhost:3000/")
```